back/engines/commercial/analysis/app/controllers/analysis/web_api/v1/tags_controller.rb

```diff
@@ -10,15 +10,26 @@ class TagsController < ApplicationController
         def index
           @tags = @analysis.tags
 
-          total_input_counts = TagCounter.new(@analysis, tags: @tags).execute
-          filtered_input_counts = TagCounter.new(@analysis, tags: @tags, filters: filter_params.to_h).execute
+          inputs_count_by_tag = TagCounter.new(@analysis, tags: @tags).counts_by_tag
+          filtered_inputs_count_by_tag = TagCounter.new(@analysis, tags: @tags, filters: filter_params.to_h).counts_by_tag
+
+          inputs_total = TagCounter.new(@analysis, tags: @tags, filters: {}).total_count
+          filtered_inputs_total = TagCounter.new(@analysis, tags: @tags, filters: filter_params.to_h).total_count
+          inputs_without_tags = TagCounter.new(@analysis, tags: @tags, filters: { tag_ids: [] }).total_count
+          filtered_inputs_without_tags = TagCounter.new(@analysis, tags: @tags, filters: filter_params.to_h.merge(tag_ids: [])).total_count
```

Comment:
These 2 lines will need to change when we change the 'without tags' filtering to use `[nil]`

Reply:
Could you take this into account already? 🙏